### PR TITLE
[ENG-2469] Upgrade pip

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

We wanted to upgrade our version of pip and found that the conflict resolver did not work as expected, so I decide to resolve our dependency conflicts bey updating our conflicting dependencies. One of those dependencies is pytest and pytest 5.0.0 outlawed certain fixture behavior, so I had to change some of those fixtures to functions. 

The testmon PR has a lot of overlap with this, because that also upgrades the conflicting dependencies.

## Changes

- updates some old dependencies to update pip and alpine. 

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify pip installs correctly and the site deploys well.

What are the areas of risk?

We are making changes to our Dockerfile.

## Documentation

tech task, no docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-2469

### CODE REVIEW

File: api_tests/users/views/test_user_actions.py [hunk 1]
```diff
@@ -18,11 +18,13 @@ class TestReviewActionFilters(ReviewActionFilterMixin):
     def url(self):
         return '/{}actions/reviews/'.format(API_BASE)
 
+    def get_expected_actions(self, all_actions, allowed_providers):
```
Comment (corbinSanders) [inline api_tests/users/views/test_user_actions.py:21]: Why was this redefined from api_tests/reviews/mixins/filter_mixins.py

Comment (Johnetordoff) [reply api_tests/users/views/test_user_actions.py:21]: Pytest 5.0.0 fully depreciated callable fixtures, so it threw and error and failed in pytest 5

**Comment by corbinSanders** [comment]
Looks good. Thanks for the clairification in the PR description

**Comment by jwalz** [comment]
So, in order to actually figure out our conflicts, you need to remove the instances of the `--use-deprecated=legacy-resolver` arg that were added in https://github.com/CenterForOpenScience/osf.io/pull/9556/files

**Review by corbinSanders — CHANGES_REQUESTED** [review]
pip version 21 was released over the weekend.

**Comment by corbinSanders** [comment]
Teamwork makes the dream work. 🚀

**Review by jwalz — APPROVED** [review]

**Pull request merged**